system/lib/pthread/proxying.c

```diff
@@ -268,6 +366,7 @@ int emscripten_proxy_async(em_proxying_queue* q,
   // Otherwise, the target thread was already notified when the existing work
   // was enqueued so we don't need to notify it again.
   if (empty) {
+    q->js_refcount++;
```

Comment:
Would it make sense to do this as part of `_emscripten_notify_proxying_queue` in JS so that all the handling of the count is done on the JS side?

Reply:
Hmm maybe, but keeping it in Wasm probably saves a few bytes of code size. Do you think the symmetry is worth it?